src/plugins/workspace/server/workspace_client.ts

```diff
@@ -248,6 +265,12 @@ export class WorkspaceClient implements IWorkspaceClientImpl {
           version: workspaceInDB.version,
         }
       );
+
+      if (newDataSources && this.uiSettings && client) {
```

Comment:
Nit: I think we can remove `newDataSources` from `if (newDataSources && this.uiSettings && client)` because this part of the code is already in `if (newDataSources)`

Reply:
Could you elaborate it more? I suppose this code is not in the block of `if (newDataSources)`.